AppCenterDataStorage/AppCenterDataStorage/Internal/Model/MSTokenResult.m

```diff
@@ -23,8 +34,69 @@ - (instancetype)initWithPartition:(NSString *)partition
     _dbCollectionName = dbCollectionName;
     _token = token;
     _status = status;
+    _expiresOn = expiresOn;
+  }
+  return self;
+}
+
+- (instancetype)initWithDictionary:(NSDictionary *)tokens {
+  self = [super init];
+  if (self) {
+    self = [[MSTokenResult alloc] initWithPartition:tokens[kMSPartition]
+                                          dbAccount:tokens[kMSDbAccount]
+                                             dbName:tokens[kMSDbName]
+                                   dbCollectionName:tokens[kMSDbCollectionName]
+                                              token:tokens[kMSToken]
+                                             status:tokens[kMSStatus]
+                                          expiresOn:tokens[kMSExpiresOn]];
   }
   return self;
 }
 
+- (instancetype)initWithString:(NSString *)tokenString {
+  self = [super init];
+  if (self) {
+    NSData *jsonData = [tokenString dataUsingEncoding:NSUTF8StringEncoding];
+    NSError *error;
+    NSDictionary *jsonDictionary = [NSJSONSerialization JSONObjectWithData:jsonData options:kNilOptions error:&error];
+    if (jsonData != nil && error == nil) {
+
+      // Create token result object.
+      self = [[MSTokenResult alloc] initWithPartition:jsonDictionary[kMSPartition]
+                                            dbAccount:jsonDictionary[kMSDbAccount]
+                                               dbName:jsonDictionary[kMSDbName]
+                                     dbCollectionName:jsonDictionary[kMSDbCollectionName]
+                                                token:jsonDictionary[kMSToken]
+                                               status:jsonDictionary[kMSStatus]
+                                            expiresOn:jsonDictionary[kMSExpiresOn]];
+    }
+  }
+  return self;
+}
+
+- (NSDictionary *)serializeToDictionary {
```

Comment:
I know it's only a private method but I find `serializeToDictionary` naming a bit confusing because serialization by definition means transform a data structure into a format (like string) that can be transmitted. Converting object to Dictionary is not serialization. Maybe `convertToDictionary` is a better naming.